config/src/config.rs

```diff
@@ -162,61 +164,96 @@ impl KeyPairs {
 
         file.write_all(&contents).expect("Error writing file");
     }
+
     // used in testing to fill the structure with test keypairs
-    pub fn load(private_keys: &TrustedPeerPrivateKeys) -> Self {
-        let network_signing_private_key = private_keys.get_network_signing_private();
+    pub fn load(private_keys: TrustedPeerPrivateKeys) -> Self {
+        let (network_signing_private_key, network_identity_private_key, consensus_private_key) =
+            private_keys.get_key_triplet();
         let network_signing_public_key = (&network_signing_private_key).into();
-        let network_identity_private_key = private_keys.get_network_identity_private();
         let network_identity_public_key = (&network_identity_private_key).into();
-        let consensus_private_key = private_keys.get_consensus_private();
         let consensus_public_key = (&consensus_private_key).into();
         Self {
-            network_signing_private_key,
+            network_signing_private_key: Some(network_signing_private_key),
             network_signing_public_key,
             network_identity_private_key,
             network_identity_public_key,
-            consensus_private_key,
+            consensus_private_key: Some(consensus_private_key),
             consensus_public_key,
         }
     }
     // getters for private keys
-    pub fn get_network_signing_private(&self) -> signing::PrivateKey {
-        self.network_signing_private_key.clone()
+    /// Beware, this destroys the private key from this NodeConfig
+    pub fn network_signing_private(&mut self) -> Option<Ed25519PrivateKey> {
+        std::mem::replace(&mut self.network_signing_private_key, None)
```

Comment:
should we zero the memory instead? I see std::mem::replace() doesn't drop values /cc @mimoo

Reply:
Ideally, yes, but:
- since the memory for those is not pinned, it may not be so useful,
- we'd be leaving the X25519 key behind in investing in this work this early,

Should we file an issue for later work? — after finishing the port to the new API, we'll have a much better ability to track **every** instance of key material, and address memory storage of all private keys in one fell swoop.